application/electron/src/services/files/service.file.recent.ts

```diff
@@ -0,0 +1,245 @@
+            Object.keys(this._subscriptions).forEach((key: string) => {
+                this._subscriptions[key].destroy();
+            });
+            resolve();
+        });
+    }
+
+    public getName(): string {
+        return 'ServiceFileRecent';
+    }
+
+    public save(file: string, size: number) {
+        const stored: IStorageScheme.IStorage = ServiceStorage.get().get();
+        const files: IStorageScheme.IRecentFile[] = stored.recentFiles.filter((fileInfo: IStorageScheme.IRecentFile) => {
+            return fileInfo.file !== file;
+        });
+        if (files.length > MAX_NUMBER_OF_RECENT_FILES) {
+            files.splice(files.length - 1, 1);
+        }
+        files.unshift({
+            file: file,
+            filename: path.basename(file),
+            folder: path.dirname(file),
+            timestamp: Date.now(),
+            size: size,
+        });
+        ServiceStorage.get().set({
+            recentFiles: files,
+        });
+        ServiceElectron.updateMenu();
+    }
+
+    public saveFilters(file: string, filters: number) {
+        const stored: IStorageScheme.IStorage = ServiceStorage.get().get();
+        const files: IStorageScheme.IRecentFilterFile[] = stored.recentFiltersFiles.filter((fileInfo: IStorageScheme.IRecentFilterFile) => {
+            return fileInfo.file !== file;
+        });
+        if (files.length > MAX_NUMBER_OF_RECENT_FILES) {
+            files.splice(files.length - 1, 1);
+        }
+        files.unshift({
+            file: file,
+            filename: path.basename(file),
+            folder: path.dirname(file),
+            timestamp: Date.now(),
+            filters: filters,
+        });
+        ServiceStorage.get().set({
+            recentFiltersFiles: files,
+        });
+        ServiceElectron.updateMenu();
+    }
+
+    public get(): Promise<IStorageScheme.IRecentFile[]> {
+        return new Promise((resolve) => {
+            const stored: IStorageScheme.IStorage = ServiceStorage.get().get();
+            const checked: IStorageScheme.IRecentFile[] = [];
+            Promise.all(stored.recentFiles.map((file: IStorageScheme.IRecentFile) => {
+                return new Promise((resolveFile) => {
+                    fs.stat(file.file, (err: NodeJS.ErrnoException | null, stats: fs.Stats) => {
+                        if (err) {
+                            return resolveFile();
+                        }
+                        file.size = stats.size;
+                        checked.push(file);
+                        resolveFile();
+                    });
+                });
+            })).then(() => {
+                if (checked.length === stored.recentFiles.length) {
+                    return resolve(checked);
+                }
+                ServiceStorage.get().set({
+                    recentFiles: checked,
+                });
+                ServiceElectron.updateMenu();
+                resolve(checked);
+            });
```

Comment:
potential "uncatched promise rejection" place. Section `catch` should be present with promises almost always.